### PR TITLE
apply cs fixes

### PR DESCRIPTION
**Pull request by dbu**

### CODE REVIEW

**Comment by dbu** [comment]
and now phpstan complains :man_facepalming: 

will look into this some more on monday.

**Comment by chregu** [comment]
Yup. Had the same issue ;) And couldn't fix it, but didn't try hard ennough.

 And thanks for looking into it.

**Comment by dbu** [comment]
"next monday" is the new "next tuesday". but looked into it now and fixed the phpstan issues.

the part about marking types with `?` when they default to `null` is relevant for PHP 8.4. that will do deprecation warnings for non nullable types that default to null.

File: src/Core/DynamicMetadata/DetectionFace.php [hunk 1]
```diff
@@ -60,6 +60,9 @@ public static function getName()
         return 'detection_face';
     }
 
+    /**
+     * @return $this
```
Comment (thePanz) [inline src/Core/DynamicMetadata/DetectionFace.php:64]: should this be `@return self` or `@return DetectionFace`?

I guess we can not update the method's signature for BC reasons?

File: src/Core/DynamicMetadata/SubjectArea.php [hunk 2]
```diff
@@ -88,6 +88,9 @@ public static function getName()
         return 'subject_area';
     }
 
+    /**
+     * @return self
```
Comment (thePanz) [inline src/Core/DynamicMetadata/SubjectArea.php]: here we used `@return self` (See above comment)

File: src/Image.php [hunk 3]
```diff
@@ -769,7 +769,6 @@ public function addAutolabels($hash, $organization = null): SourceImage
      * @param string[]    $languages
      * @param string      $hash
      * @param string|null $organization
-     * @param array       $options, for example ['force' => true] to recreate
```
Comment (thePanz) [inline src/Image.php:772]: Should we add instead:
```php
 @param array{force?: bool} $options
```

File: src/LocalImage/AbstractLocalImage.php [hunk 4]
```diff
@@ -21,14 +21,14 @@ abstract class AbstractLocalImage
...
-    protected $context = null;
+    protected $context;
 
     /**
      * The rokka hash from the rokka API.
      *
      * @var string|null
      */
-    protected $rokkaHash = null;
+    protected $rokkaHash;
```
Comment (thePanz) [inline src/LocalImage/AbstractLocalImage.php:31]: this is not initialized in the constructor, we should add its initialization too

File: src/UriHelper.php [hunk 5]
```diff
@@ -160,13 +160,15 @@ public static function decomposeUri(UriInterface $uri)
...
-            preg_match('#^/'.$stackPattern.'/'.$hashPattern.'.'.$formatPattern.'$#', $path, $matches) ||
+            || preg_match('#^/'.$stackPattern.'/'.$hashPattern.'.'.$formatPattern.'$#', $path, $matches)
             // remote_path with seo-filename
-            preg_match('#^/'.$stackPattern.'/'.$pathPattern.'/'.$filenamePattern.'\.'.$formatPattern.'$#', $path, $matches) ||
+            || preg_match('#^/'.$stackPattern.'/'.$pathPattern.'/'.$filenamePattern.'\.'.$formatPattern.'$#', $path, $matches)
             // remote_path without seo-filename
-            preg_match('#^/'.$stackPattern.'/'.$pathPattern.'.'.$formatPattern.'$#', $path, $matches)) {
+            || preg_match('#^/'.$stackPattern.'/'.$pathPattern.'.'.$formatPattern.'$#', $path, $matches)
+        ) {
+            // @phpstan-ignore-next-line
```
Comment (thePanz) [inline src/UriHelper.php]: we should upgrade to phpstan v1.11 and ignore specific errors here (they introduced error identifiers in v1.11)

**Review by thePanz — CHANGES_REQUESTED** [review]
Added a few comments, we should adapt the changes

File: src/Core/DynamicMetadata/DetectionFace.php (hunk 1 above)
Comment (dbu) [reply src/Core/DynamicMetadata/DetectionFace.php:64]: according to the phpdoc [psr-5](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#keyword) (only in draft state, but i think still an indication) this is the correct type here. see 16. item in the list.

File: src/Core/DynamicMetadata/SubjectArea.php (hunk 2 above)
Comment (dbu) [reply src/Core/DynamicMetadata/SubjectArea.php]: fixing to `$this` to be more precise

File: src/Image.php (hunk 3 above)
Comment (dbu) [reply src/Image.php:772]: indeed. not sure why i just deleted this, as it provides information

File: src/LocalImage/AbstractLocalImage.php (hunk 4 above)
Comment (dbu) [reply src/LocalImage/AbstractLocalImage.php:31]: that is the cs fixer removing redundant code. when there is no type, it is automatically initialized to `null`. when we add property types, we will need to re-add the ` = null` but we should only do that in a major version.

File: src/UriHelper.php (hunk 5 above)
Comment (dbu) [reply src/UriHelper.php]: a, nice. and with `phpstan -v` one gets to see the rules

**Review by thePanz — APPROVED** [review]

**Pull request merged**